modules/service/src/main/scala/lucuma/odb/service/CalibrationsService.scala

```diff
@@ -280,65 +315,17 @@ object CalibrationsService {
             .map(spectroPhotometricTargets(referenceInstant))
           gsTgt = bestTarget(gscoords, tgts)
           gnTgt = bestTarget(gncoords, tgts)
-          gnls <- session.execute(Statements.selectGmosNorthLongSlitConfigurations(false))(pid)
-          gsls <- session.execute(Statements.selectGmosSouthLongSlitConfigurations(false))(pid)
-          _    <- generateCalibrations(pid, gnls, gsls, gnTgt, gsTgt).whenA(gnls.nonEmpty || gsls.nonEmpty)
+          (scienceGmosNLS, scienceGmosSLS) <- uniqueConfiguration(pid, ObservationSelection.Science)
+          (calibGmosNLS, calibGmosSLS)     <- uniqueConfiguration(pid, ObservationSelection.Calibration)
+          gnls = scienceGmosNLS.diff(calibGmosNLS)
+          gsls = scienceGmosSLS.diff(calibGmosSLS)
+          _                                <- generateCalibrations(pid, gnls, gsls, gnTgt, gsTgt).whenA(gnls.nonEmpty || gsls.nonEmpty)
```

Comment:
this is probably still too naive approach but basically I look for all the existing different configurations for science, and I create new calibrations for the configurations not already in a calibration.

There is some fragility here that I need to resolve before putting this into production. If the calibrations exist but they don't produce a configuration (e.g. if the target is not appropriate) this can produce an endless stream of configurations.

Reply:
What happens to existing calibrations for which there is no longer a matching science observation?

Reply:
yes, that's also pending, we need to delete them. There is a shortcut task for that